### PR TITLE
Add test_helpers module and remove _build_one_column_dataset"

### PR DESCRIPTION
**Pull request by TomGoBravo**
This PR
* Adds test_helpers module so it is easier to import without creating import cycles
* Adds `build_default_region_dataset`, a thin wrapper around `build_dataset`
* Removes `_build_one_column_dataset` which is easily replaced by `build_default_region_dataset`

### CODE REVIEW

File: test/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    def __init__(
+        self, ts_list, *, provenance: str = "",
+    ):
+        super().__init__(ts_list)
+        self.provenance = provenance
+
+
+def build_dataset(
+    metrics: Mapping[Region, Mapping[FieldName, Union[Sequence[float], TimeseriesLiteral]]],
```
Comment (ghop02) [inline test/test_helpers.py:37]: I think, `metrics_by_region` might be more helpful as to what the mapping is

File: test/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    *,
+    start_date="2020-04-01",
+    dropna=True,
+) -> timeseries.MultiRegionDataset:
+    """Returns a dataset for multiple regions and metrics. Each sequence of values represents a
+    timeseries metric with identical length. provenance information can be set for a metric by
+    using a TimeseriesLiteral. Timeseries without any real values are dropped.
```
Comment (ghop02) [inline test/test_helpers.py:44]: Can you add args here?

File: test/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            dtype="str",
+            index=provenance_index,
+            name=PdFields.PROVENANCE,
+        )
+        dataset = dataset.add_provenance_series(provenance_series)
+
+    return dataset
+
+
+def build_default_region_dataset(
```
Comment (ghop02) [inline test/test_helpers.py:88]: how would you feel about: 
```
def build_single_region_dataset(metrics: Mapping..., *, region=DEFAULT_REGION, dropna=True)
```

lets you customize the region if you want

**Review by ghop02 — APPROVED** [review]
Looks good! couple suggestions on naming

File: test/test_helpers.py (hunk 1 above)
Comment (TomGoBravo) [reply test/test_helpers.py]: I thought the type make it fairly clear. Changed to `metrics_by_region_then_field_name`

File: test/test_helpers.py (hunk 2 above)
Comment (TomGoBravo) [reply test/test_helpers.py]: done

File: test/test_helpers.py (hunk 3 above)
Comment (TomGoBravo) [reply test/test_helpers.py:91]: I'm down for that YAGNI.

**Pull request merged**